wallet-new/server/Cardano/Wallet/API/V1/Swagger/Example.hs

```diff
@@ -0,0 +1,58 @@
+module Cardano.Wallet.API.V1.Swagger.Example where
+
+import           Universum
+
+import           Cardano.Wallet.API.Response
+import           Cardano.Wallet.API.V1.Types
+
+import           Test.QuickCheck (Arbitrary (..), Gen, listOf1)
+
+class Arbitrary a => Example a where
+    example :: Gen a
+    example = arbitrary
+
+instance {-# OVERLAPS #-} Arbitrary a => Example a
```

Comment:
This looks a bit strange, why we need such empty, overlappable instance?

Reply:
With it we say "Everything that has Arbitrary instance also has Example instance". So it says that every Arbitrary is also Example. If we remove it we have to explicitly define all instances... for example:
```
/home/akegalj/projects/serokell/cardano-sl/wallet-new/server/Cardano/Wallet/API/V1/Swagger/Example.hs:30:27: error:
    • No instance for (Example
                         cardano-sl-core-1.0.3:Pos.Core.Common.Types.Coin)
```

and dozens more (<20). So we would have to define:
```
instance Example Coin where
    example = arbitrary
```
explicitly and with `{-# OVERLAPS #-}` this is implicit. 

So I believe it is a trade off between the above boilerplate (I don't mind writing it though).

Do you think I should remove it and write instances manually ?

Reply:
In the meantime I have changed `OVERLAPS` with `OVERLAPPABLE` as it seems safer https://ghc.haskell.org/trac/ghc/ticket/9242#comment:30 . 

I will implement it without pragmas by explicitly writing instances (which is not hard :) ) if we decide to go this way (ie, it is 5 mins of work).

Reply:
Let me write Pros and Cons of both approaches on Monday and then we will make a final decision.

Reply:
I found this interesting approach - https://kseo.github.io/posts/2017-02-05-avoid-overlapping-instances-with-closed-type-families.html

Reply:
@ksaric Interesting. Let me read it and try playing with it on Monday. Ty

Reply:
I got interested in this issue, in the end, I ended up writing up the code, so I added a commit. You can remove it, I'm pretty sure `OVERLAPPABLE` is going to be clearer.

Reply:
I'm aware of that blog post and I'm sure we can go via that route, although I'm just skeptical we need to bring cannons on the table (e.g. type families) for such a simple scenario, which can be solved by literally 30 seconds of vim macros to write a bunch of one-lines like this:

```
instance Example Foo
instance Example Bar
```

This will work without defining the body of the instance due to the default implementation. Boilerplat-y? Yes. Haskell-98? Yes 😁 

I still think using closed type families for avoid overlappable instances is a good trick to remember and keep in our arsenal, but I don't think the problem at hand requires them 😉

Reply:
Fixed, I believe we can merge after CI